Update projects.php

SweaterWeather opened the pull request:
Updated projects.php
- page now pulls image urls from the database
- page now pulls project creator screen name from the database
- page now pulls project title from the database
- clicking on a project's thumbnail will redirect you to the relevant project page

LoganRBrown committed:
Projects Page is reworked

Changed the projects.php page to run on php rather than html.

LoganRBrown committed:
Projects Page is reworked

Changed the projects.php page to run on php rather than html.

Commit:
.php: Auto stash before rebase of "origin/Logans_work"

Commit:
.php: Auto stash before rebase of "origin/Logans_work"

Commit:
Finished adding the additional hover info

Pretty much exactly what it sounds like.  Now hovering over any image will display placeholder text for the project's info.

Commit:
Finished adding the additional hover info

Pretty much exactly what it sounds like.  Now hovering over any image will display placeholder text for the project's info.

Commit:
Final tweek

Reposistioned a thing.

Commit:
Final tweek

Reposistioned a thing.

Commit:
Merge remote-tracking branch 'origin/update_projects.php' into update_projects.php

Commit:
Hooked Projects.php to the database

Projects.php will now populate itself with relevant content from the database, including images, creator details, and project titles.  Clicking on these will redirect you to the proper project page.

nickworks commented:
I know I'm being a little nit-picky, but here is the list of changes / cleanups I'd like to see:

- Remove **project.php~Stashed changes**.
- Remove **project.php~Updated upstream**.
- Use a foreach instead of a while (line 58)
- If there is no image url, let's show something else (a dark gray background?)
- The **arrayRandom()** function seems unnecessary. Can you combine it with **assignCssClass()** ?
- Also, let's try these odds: 70% normal, 10% big, 10% horizontal, 10% vertical
- Use overflow:hidden on the &lt;a&gt; to crop the overlay in the box.
- &lt;h1&gt; &lt;h2&gt; and &lt;h3&gt; are block-level elements, which should NOT be contained within inline elements.  So this code is technically invalid HTML. Please replace these with spans and style with classes.
- Since you only need one image per project, try a single SQL query to join the project_images table:
```SQL
SELECT p.id, p.title, u.alias, i.url FROM projects p, users u, project_imgs i WHERE u.id = p.user_id AND i.project_id = p.id GROUP BY p.id ORDER BY p.id DESC
```

Commit:
Removed stashed changes

Commit:
Removed block elements from the grid

Commit:
Refactored code

Now only one SQL request is made and eliminated arrayRandom

Commit:
Implimented the last of Nick's changes

Fixed assignCssClass(), replaced the block elements with spans, added a default image for missing images, and finally styled it all.

Commit:
Removed empty space

Also tweaked the overflow:hidden on anchor tags
